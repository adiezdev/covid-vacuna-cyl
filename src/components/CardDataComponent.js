import React from 'react'

export const CardDataComponent = ({ sumaReduce,title}) => {

    let { total , groupByType} = sumaReduce;
    
    const { AstraZeneca, Moderna ,Pfizer } = {...groupByType}

    return (
        <div className="maincard">
            <header className="t-center-big">{ title }</header>
            <div className="t-center-big">{total}</div>
            {
                !AstraZeneca || !Moderna || !Pfizer ? 
                <div></div>:
                <ul>
                <li className="t-center-midle">{Pfizer.vacuna}</li>
                <li className="t-center-midle">{Moderna.vacuna}</li>
                <li className="t-center-midle">{AstraZeneca.vacuna}</li>
                </ul>
            }
        </div>
    )
}
