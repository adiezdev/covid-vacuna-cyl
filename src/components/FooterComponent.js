import React from 'react'

/*
* Componente project information
*/
export const FooterComponent = () => {
    console.log("Â¡Te pillÃ©!ðŸ˜œ\n%c+","font-size: 12px; padding: 30px 200px; line-height: 200px; background: url(https://media.giphy.com/media/6ra84Uso2hoir3YCgb/giphy.gif); color: transparent;");
    return (
        <>
        <footer className="footer">
            <p>
                Desarrollado por <a href='https://github.com/adiezdev'>Alejandro DÃ­ez</a> inspirado en el proyecto <a href='https://covid-vacuna.app/'>Covid-vacuna</a>
            </p>
            <p>
                Si quieres mi proyecto <a href='https://github.com/adiezdev/covid-vacuna-cyl'>aquÃ­ estÃ¡</a>
            </p>
            <div><a href='https://www.linkedin.com/in/adilosa95/'>Linkedin</a> | <a href='https://github.com/adiezdev'>Github</a> | <a href='https://dev.to/adilosa95'>Dev</a></div>
        </footer>
        </>
    )
}
