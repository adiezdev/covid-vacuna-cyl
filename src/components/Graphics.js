import React from 'react'

import * as d3  from "d3";

import { useFetch } from '../hooks/useFetch';
import { getPersonasVacunadas } from '../selectors/personas-vacunadas/getPersonasVacunadas';


export const Graphics = () => {
    const { data: personasvacunadasproprovincia , isLogin } = useFetch( 3000 , 'personas-vacunadas-covid')
    const { totalProgresivo } =  getPersonasVacunadas( personasvacunadasproprovincia )
    const svg = d3.select(".grap")

    if(!isLogin && totalProgresivo)
    {
        const parseDate = () => d3.timeParse("%Y-%m-%d")
        //Add X
        const margin = {
            top: 20,
            right: 30,
            bottom: 30,
            left: 40
          }
          const yAxis = (g) => g
                    .attr("transform", `translate(${margin.left},0)`)
                    .call(d3.axisLeft(y))
                    .call(g => g.select(".domain").remove())
                    .call(g => g.select(".tick:last-of-type text").clone()
                        .attr("x", 3)
                        .attr("text-anchor", "start")
                        .attr("font-weight", "bold")
                        .text("$ Close"))
          const y = d3.scaleLinear()
          .domain([0, d3.max(totalProgresivo, d => d.fields.dosis_administradas)]).nice()
          .range([700 - margin.bottom, margin.top])

          const xAxis = g => g
          .attr("transform", `translate(0,${700 - margin.bottom})`)
          .call(d3.axisBottom(x).ticks(1000 / 80).tickSizeOuter(0))

          const x = d3.scaleUtc()
          .domain(d3.extent(totalProgresivo, d => (d3.timeParse("%Y-%m-%d")(d.fields.fecha))))
          .range([margin.left, 1300 - margin.right])

          const line = d3.line()
            .defined(d => !isNaN(d.fields.dosis_administradas))
            .x(d => x((d3.timeParse("%Y-%m-%d")(d.fields.fecha))))
            .y(d => y(d.fields.dosis_administradas))

           svg.append("g")
                .call(yAxis)
            
            
           svg.append("g")
           .call(xAxis)


           svg.append("path")
           .datum(totalProgresivo)
           .attr("fill", "none")
           .attr("stroke", "steelblue")
           .attr("stroke-width", 1.5)
           .attr("d", line); 
    }    
    return (
        <div>
            <svg 
            className="centerItem"
            width="100%"
            height="170vh"
            >
                <g 
                className="grap"
                transform="translate(100 , 100)">

                </g>
            </svg>
        </div>
    )
}
