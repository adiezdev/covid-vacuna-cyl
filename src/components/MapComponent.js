import React from 'react'
import ReactTooltip from 'react-tooltip';
import data from '../data/cyl.geo.json'

import * as d3  from "d3";

import { useWindowSize } from '../hooks/useWindowSize';
import { useFetch } from '../hooks/useFetch';
import { getPersonasVacunadas } from '../selectors/personas-vacunadas/getPersonasVacunadas';
import { getVacunasRecibidas } from '../selectors/vacunas-recibidas/getVacunasRecibidas';


export const MapComponent = () => {
        

    const { data: personasvacunadasproprovincia  } = useFetch( 1000 , 'personas-vacunadas-covid')
    const { ...groupByDosisProvincia } =  getPersonasVacunadas( personasvacunadasproprovincia )

    const { data: vacunasrecibidas  } = useFetch(27 , 'vacunas-recibidas-covid')
    const { groupByVacunasProvincias } = getVacunasRecibidas( vacunasrecibidas )

    const  {windowSize}  = useWindowSize();
    //Projection to map
    const projection = d3.geoMercator()
                        .scale([windowSize.with > 722 ? 6800 :  (950 * windowSize.with)/100])
                        .center([ windowSize.with > 435 ?  (-0.45 * windowSize.with)/100 : windowSize.with > 360 ? (-0.05* windowSize.with)/100  : ( 0.20 * windowSize.with)/100  , windowSize.with > 500 ? 42 : 41.2])
    //Paint var to json
    const pathGenerator = d3.geoPath().projection(projection)
    
    return (
        <>     
        <svg          
            className="cylmap centerItem"
        >
         {
            data.features.map((d,i)=>
            <path
                key={`path${i}`}
                d={pathGenerator(d)}
                className="provincies"
                data-tip={d.properties.name} //We pass the province to the toolip
                data-for='data-province'
            />
            )
        }
        </svg>
        <ReactTooltip id='data-province' aria-haspopup='true' getContent={(dataTip)=>
            !groupByDosisProvincia[dataTip] || !groupByVacunasProvincias[dataTip]
            ?
            dataTip
            :
            <div className="t-small">
                <div className='t-small t-center'>{dataTip}</div>
                <p> - Dosis Administradas: {groupByDosisProvincia[dataTip].vacuna}</p>
                <p> - Dosis Recibidas: {groupByVacunasProvincias[dataTip].vacuna}</p>
            </div>
        }/>
        </>
    )
}
