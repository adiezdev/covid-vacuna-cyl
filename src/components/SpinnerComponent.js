import React from 'react'
import PropTypes from 'prop-types'
/**
 * 
 * Component Spinner
 * @returns 
 */
export const SpinnerComponent = ({otherclass = 'p-inline-block'}) => {
    return (
        <div className={`spinner ${otherclass}`}>
        </div>
    )
}

SpinnerComponent.propTypes ={
    otherclass: PropTypes.string
}