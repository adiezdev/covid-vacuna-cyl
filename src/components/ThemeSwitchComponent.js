import React from 'react'

export const ThemeSwitchComponent = () => {

    return (
        <div className='switch'>
            <div
                className='themeLight'
                role='img'
            >
            🌞
            </div>
            <div
                className='themeSystme'
                role='img'
            >
            💻
            </div>
            <div
                className='themeDark'
                role='img'
            >
            🌚
            </div>
        </div>
    )
}
