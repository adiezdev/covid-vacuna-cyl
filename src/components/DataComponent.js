import React, { useMemo } from 'react'
import { CardDataComponent } from './CardDataComponent'
import { useFetch } from '../hooks/useFetch'
import { getTotalPersonasVacunadas } from '../selectors/personas-vacunadas/getTotalPersonasVacunadas'
import { getTotalCicloCompleto } from '../selectors/personas-vacunadas/getTotalCicloCompleto'
import { getTotalVacunasRecibidas } from '../selectors/vacunas-recibidas/getTotalVacunasRecibidas'

export const DataComponent = () => {
    const { data: personasvacunadas} = useFetch('https://analisis.datosabiertos.jcyl.es/api/records/1.0/search/?dataset=personas-vacunadas-covid&q=&rows=1000') || {}
    
    const { data: vacunasrecibidas, isLogin} = useFetch('https://analisis.datosabiertos.jcyl.es/api/records/1.0/search/?dataset=vacunas-recibidas-covid&rows=1000') || {}


    const informations = [
        {
            'id':  0,
            'title':'Total de personas vacunadas',
            'sumaReduce': useMemo(() => getTotalPersonasVacunadas(personasvacunadas ), [personasvacunadas]) 
        },
        {
            'id':  1,
            'title':'Total de personas vacunadas por el ciclo completo',
            'sumaReduce':  useMemo(() => getTotalCicloCompleto(personasvacunadas ), [personasvacunadas]) 
        },
        {
            'id':  2,
            'title':'Total de vacunas recibidas',            
            'sumaReduce': useMemo(() => getTotalVacunasRecibidas(vacunasrecibidas ), [vacunasrecibidas]) 
        },
    ]
    return (
        <>
            <h1 className="t-center-big">Datos Recibidos</h1>
            <div className="center-item-wrap">
                {
                    !isLogin ?
                    informations.map( information => (
                        <CardDataComponent
                            key={information.id}
                            sumaReduce={information.sumaReduce}
                            title={information.title}
                        />
                    ))
                    : <div className="spinner p-inline-blok"></div> 
                }
            </div>
        </>
        
    )
}