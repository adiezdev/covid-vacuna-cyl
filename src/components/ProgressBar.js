import React from 'react'
import { useFetch } from '../hooks/useFetch'
import { useIncrement } from '../hooks/useIncrement'
import { getPersonasVacunadas } from '../selectors/personas-vacunadas/getPersonasVacunadas'
import { SpinnerComponent } from './SpinnerComponent'

export const ProgressBar = React.memo(() => {
    const { data , isLogin } = useFetch( 1 , 'personas-vacunadas-covid','TotalCyL')
    const {porcientoPoblacionVacunada} =getPersonasVacunadas( data )
    const {count} = useIncrement(porcientoPoblacionVacunada?.toFixed(1))
    return (
        <> 
        {
            !isLogin 
            ?
            <div className="container-progress">
                <div 
                className="filler-progress" 
                style={
                    {width: `${porcientoPoblacionVacunada}%`}
                }
                onAnimationStart="grwoup 2s ease-in-out;"
                >
                    <span 
                    className="label-progres t-midle"
                    role="progressbar"
                    aria-valuenow={porcientoPoblacionVacunada}
                    aria-valuemin="0"
                    aria-valuemax="100"
                    >{count}%</span>
                </div>
            </div> 
            : 
            <div className="grap center-item-wrap-grap">
            <SpinnerComponent/>
            </div>
        }
      
        </>
    )
})