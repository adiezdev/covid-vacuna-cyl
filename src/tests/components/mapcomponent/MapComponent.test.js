import Enzyme, { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'
;
import { MapComponent } from "../../../components/MapComponent";

import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Pruebas unitarias parel componente <MapComponent/>',()=>{
    
    test('Renderizar correctamente', () => {
        const wrapper = shallow(<MapComponent/>)
        expect(shallowToJson(wrapper)).toMatchSnapshot()

    })
    
})