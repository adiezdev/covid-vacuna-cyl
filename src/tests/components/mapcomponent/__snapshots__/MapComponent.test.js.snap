// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias parel componente <MapComponent/> Renderizar correctamente 1`] = `
<Fragment>
  <svg
    className="cylmap centerItem"
  >
    <path
      className="provincies"
      d="M480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250Z"
      data-for="data-province"
      data-tip="Ávila"
      key="path0"
      style={
        Object {
          "fill": "green",
        }
      }
    />
    <path
      className="provincies"
      d="M480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250ZM480,250L480,250L480,250L480,250L480,250ZM480,250L480,250L480,250L480,250ZM480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250ZM480,250L480,250L480,250L480,250ZM480,250L480,250L480,250L480,250ZM480,250L480,250L480,250L480,250Z"
      data-for="data-province"
      data-tip="Burgos"
      key="path1"
      style={
        Object {
          "fill": "green",
        }
      }
    />
    <path
      className="provincies"
      d="M480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250ZM480,250L480,250L480,250L480,250Z"
      data-for="data-province"
      data-tip="León"
      key="path2"
      style={
        Object {
          "fill": "green",
        }
      }
    />
    <path
      className="provincies"
      d="M480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250ZM480,250L480,250L480,250L480,250ZM480,250L480,250L480,250L480,250L480,250L480,250ZM480,250L480,250L480,250L480,250ZM480,250L480,250L480,250L480,250ZM480,250L480,250L480,250L480,250ZM480,250L480,250L480,250L480,250Z"
      data-for="data-province"
      data-tip="Palencia"
      key="path3"
      style={
        Object {
          "fill": "green",
        }
      }
    />
    <path
      className="provincies"
      d="M480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250Z"
      data-for="data-province"
      data-tip="Salamanca"
      key="path4"
      style={
        Object {
          "fill": "green",
        }
      }
    />
    <path
      className="provincies"
      d="M480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250Z"
      data-for="data-province"
      data-tip="Segovia"
      key="path5"
      style={
        Object {
          "fill": "green",
        }
      }
    />
    <path
      className="provincies"
      d="M480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250ZM480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250ZM480,250L480,250L480,250L480,250L480,250Z"
      data-for="data-province"
      data-tip="Valladolid"
      key="path6"
      style={
        Object {
          "fill": "green",
        }
      }
    />
    <path
      className="provincies"
      d="M480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250Z"
      data-for="data-province"
      data-tip="Zamora"
      key="path7"
      style={
        Object {
          "fill": "green",
        }
      }
    />
    <path
      className="provincies"
      d="M480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250L480,250Z"
      data-for="data-province"
      data-tip="Soria"
      key="path8"
      style={
        Object {
          "fill": "green",
        }
      }
    />
  </svg>
  <ReactTooltip
    aria-haspopup="true"
    clickable={false}
    getContent={[Function]}
    id="data-province"
    insecure={true}
    resizeHide={true}
    wrapper="div"
  />
</Fragment>
`;
